### ISSUE TITLE
add_role modifies redirect when there are errors in parent model, but only with STI model names

### ISSUE THREAD

**Issue by luisalima** [RolifyCommunity/rolify#117]
So, I'm _extremely_ new to Rails, so I'll try to explain myself as clearly as possible. Please apologize for any confusions in my speech. I'm using rolify 3.2.0 and devise 2.1.2.

I have a model Employer, which has_one :user (from Devise). (I am including the "rolify" in the table "User" and I assume that that generates a has_and_belongs_to_many relationship between User and Role).

In the employers_controller, in create, I assign a role to the current_user (or any other user for that matter). When I do that _and_ there are errors in my form for new employer (for example, a validation for required "name" and name isn't filled), the redirect is modified to an action "edit" with a controller with the name of the role that I am adding.

I have STI in my model -- Employer is one of the parent models. If I use any of the STI model names for my roles, this happens. This doesn't happen with any other name. It took me a while to find out about this :-)

So, to sum it up: current_user.add_role :any_name_but_sti_models, the result is correct (the errors get displayed in the form, redirect is to the "new" action of the controller.

If I use current_user.add_role :employer (or ANY of the sti model names), the result is:
:No route matches {:action=>"edit", :controller=>"employers", :id=>nil}

(I also tried with current_user.roles << Role.where(:name => :employer); the result is exactly the same).

Puzzled...
